typescript-generator/index.js

```diff
@@ -237,6 +224,9 @@ function cleanPropertyName (name) {
 
 function isSpecialInterface (type) {
   if (Array.isArray(type.attachedBehaviors)) {
+    // assume types ending with Base are abstract and are not the ones doing the implements
+    if (type.name.name.endsWith('Base'))
```

Comment:
@swallez @delvedor Thoughts on making this explicit in the model? e.g:

Any type that ends with `Base` gets `abstract: true` on type unless `@not_abstract` is specified on the type for edgecases?

Also happy to flip that and explicitly mark classes with `@abstract` in the input spec.

Reply:
I'd rather be explicit in the input spec rather than rely on implicit naming conventions.

Reply:
++ aggreed, this should be easier once https://github.com/elastic/elastic-client-generator/pull/1634lands